### PR TITLE
Improving CMakeLists

### PR DESCRIPTION
**Pull request by bekorn**
Issue: https://github.com/GPUOpen-Tools/compressonator/issues/261

Major Changes
- Makes cmake -DOPTION_BUILD_*=ON/OFF arguments have an effect

Minor Changes
- Improves the readability and utilization of CMake
- [TODO] Update documentation
This page https://compressonator.readthedocs.io/en/latest/build_from_source/build_instructions.html
- [TODO] Pick default values for options

### CODE REVIEW

**Comment by bekorn** [comment]
Sorry that I didn't explain myself. I'm not familiar with the workflow of open source contribution. That is why I created a pull request "template", waiting for approval. As this had become a non-template, should I take it as approval? Is my solution doable?

Also, I don't think that I can pick the defaults by myself since I don't know the user base or common use case of the project. I will need your guidance on that.

**Comment by denislevesqueAMD** [comment]
@bekorn I will be taking a look at this and let you know if I spot anything that should be changed

**Review by denislevesqueAMD — CHANGES_REQUESTED** [review]
I still need to download your changes on my local machine and test it, but I have some initial comments on the pull request.

File: external/CMakeLists.txt
Comment (denislevesqueAMD) [inline external/CMakeLists.txt:67]: This line is incorrect. The option is named `OPTION_CMP_OPENCV`.

**Comment by denislevesqueAMD** [comment]
@NPCompress This pull request should be closed. I made a new pull request, #267, that fixes the issue more thoroughly.

**Comment by NPCompress** [comment]
@bekorn Thank you for submission #262, with the help of @denislevesqueAMD a new pull request #267 with your changes and updates will be merged and this one will be closed.
We encourage you and others to continue to request other pull request to better serve our community members. Thanks again for your contribution.

**Pull request closed without merging**
